### PR TITLE
test: scan testing

### PR DESCRIPTION
**Pull request by pwltr**
### Description

- Add scantest
- Add universal link support
- Add deeplink support for `bitkit:`
- Improve types in `scanner.ts`

### Notes

No integration with server yet.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [ ] Unit test
- [x] No test

### QA Notes

Use `npx uri-scheme open "bitkit:scantest-xxxxxxxx" --ios` and `npx uri-scheme open "xxxxxxxx" --ios` to test (universal link may not work depending on teamId set in project file).

### CODE REVIEW

**Review by MiguelMedeiros — COMMENTED** [review]
tested ACK.

**Review by MiguelMedeiros — APPROVED** [review]

File: src/screens/TreasureHunt/Chest.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}: TreasureHuntScreenProps<'Chest'>): ReactElement => {
+	const { remoteBalance } = useLightningBalance(false);
+	const { id } = useSelector((state) => {
+		return viewControllerSelector(state, 'treasureHunt');
+	});
+
+	useBottomSheetBackPress('treasureHunt');
+
+	console.log({ id });
+	console.log({ remoteBalance });
```
Comment (coreyphillips) [inline src/screens/TreasureHunt/Chest.tsx:30]:
```suggestion
```

**Review by coreyphillips — APPROVED** [review]
LGTM!

File: src/screens/TreasureHunt/Chest.tsx (hunk 1 above)
Comment (MiguelMedeiros) [reply src/screens/TreasureHunt/Chest.tsx:30]: Hey @coreyphillips, this console.log is just for debugging purposes. @pwltr split this PR into two, and he'll remove it when he adds the API calls. No need to worry about it!

Comment (pwltr) [reply src/screens/TreasureHunt/Chest.tsx:30]: Yes exactly I submitted this PR as a checkpoint to make it easier to review. Just the design and url handling for now. Server will come in another PR and the logs will be useful.

@MiguelMedeiros As a side note I fixed the bug with the weird transition that you saw but will include that in the next PR as well.

**Pull request merged**